drivers/flash/Kconfig.simulator

```diff
@@ -0,0 +1,158 @@
+# Kconfig - Flash simulator config
+#
+# Copyright (c) 2018 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+menuconfig FLASH_SIMULATOR
+	bool
+	prompt "Flash simulator"
+	default n
+	depends on FLASH
+	select STATS
+	select STATS_NAMES
+	select FLASH_HAS_PAGE_LAYOUT
+	select FLASH_HAS_DRIVER_ENABLED
+	help
+	  Enable the flash simulator.
+
+if FLASH_SIMULATOR
+
+config FLASH_SIMULATOR_BASE_ADDR
+	hex
+	prompt "Base address of the flash"
+	default 0x00
+	help
+	 The base address of the flash.
+
+config FLASH_SIMULATOR_ERASE_UNIT
+	int
+	prompt "Erase unit size in bytes"
+	default 4096
+	range 1 65536
+	help
+	 The smallest area of flash memory that can be indipendently erased.
+
+config FLASH_SIMULATOR_PROG_UNIT
+	int
+	prompt "Program unit size in bytes"
+	default 4
+	range 1 4096
+	help
+	 The smallest area of flash memory that can be indipendently programmed.
+
+config FLASH_SIMULATOR_FLASH_SIZE
+	int
+	prompt "Flash size in erase units"
+	default 4
+	range 1 65536
+	help
+	 The size of the simulated flash area expressed in erase units.
+
+config FLASH_SIMULATOR_SIMULATE_FAILURES
+	bool
+	prompt "Enable failure simulation"
+	help
+	 This setting enables simulation of various API and hardware failures,
+	 which can be used for testing.
+
+config FLASH_SIMULATOR_MAX_ERASE_CYCLES
+	int
+	prompt "Maximum number of erase cycles"
+	default 10000
+	depends on FLASH_SIMULATOR_SIMULATE_FAILURES
+	help
+	 Maximum number of erase cycles after which the flash can
+	 no longer be reliably erased.
+
+config FLASH_SIMULATOR_READ_API_FAILURE_RATE_PM
+	int
+	prompt "Read API failure rate ‰ (per mille)"
+	default 0
+	range 0 1000
+	depends on FLASH_SIMULATOR_SIMULATE_FAILURES
+	help
+	 This setting controls the read API failure rate in units of 1/1000.
+	 This setting is intended for testing.
+
+config FLASH_SIMULATOR_WRITE_API_FAILURE_RATE_PM
+	int
+	prompt "Write API failure rate ‰ (per mille)"
+	default 0
+	range 0 1000
+	depends on FLASH_SIMULATOR_SIMULATE_FAILURES
+	help
+	 This setting controls the write API failure rate in units of 1/1000.
+	 This setting is intended for testing.
+
+config FLASH_SIMULATOR_ERASE_API_FAILURE_RATE_PM
+	int
+	prompt "Erase API failure rate ‰ (per mille)"
+	default 0
+	range 0 1000
+	depends on FLASH_SIMULATOR_SIMULATE_FAILURES
+	help
+	 This setting controls the erase API failure rate in units of 1/1000.
```

Comment:
thinks need add info what the API failure mean - that the driver refuse action right after was called, does not modify memory content and returns an error code.

Reply:
I agree, I will make this text more informative.